libyul/optimiser/ExpressionInliner.cpp

```diff
@@ -52,21 +51,28 @@ void ExpressionInliner::visit(Expression& _expression)
 	if (_expression.type() == typeid(FunctionCall))
 	{
 		FunctionCall& funCall = boost::get<FunctionCall>(_expression);
+		if (!m_inlinableFunctions.count(funCall.functionName.name))
+			return;
+		FunctionDefinition const& fun = *m_inlinableFunctions.at(funCall.functionName.name);
 
-		bool movable = boost::algorithm::all_of(
-			funCall.arguments,
-			[=](Expression const& _arg) { return SideEffectsCollector(m_dialect, _arg).movable(); }
-		);
-		if (m_inlinableFunctions.count(funCall.functionName.name) && movable)
+		map<YulString, Expression const*> substitutions;
+		for (size_t i = 0; i < funCall.arguments.size(); i++)
 		{
-			FunctionDefinition const& fun = *m_inlinableFunctions.at(funCall.functionName.name);
-			map<YulString, Expression const*> substitutions;
-			for (size_t i = 0; i < fun.parameters.size(); ++i)
-				substitutions[fun.parameters[i].name] = &funCall.arguments[i];
-			_expression = Substitution(substitutions).translate(*boost::get<Assignment>(fun.body.statements.front()).value);
-
-			// TODO Add metric! This metric should perform well on a pair of functions who
```

Comment:
After this pull request, I think the code size will only increase linearly every time we apply ExpressionInliner. (#6866 increases exponentially). So probably fine to remove this TODO?

Reply:
Isn't the metric essentially a boolean `refs <= 1 || cost <= 1` with this PR?

Reply:
Yes. But maybe @chriseth still want something more precise in the future? In that case, this TODO should not be removed